src/main/java/graphql/nadel/engine/NadelContext.java

```diff
@@ -0,0 +1,42 @@
+package graphql.nadel.engine;
+
+import graphql.Internal;
+
+import java.util.UUID;
+
+/**
+ * We sue a wrapper Nadel context object over top of the caller supplied one'
+ * so that we can get access to Nadel specific context properties in
+ * a type safe manner
+ */
+@Internal
+public class NadelContext {
+    final Object callerSuppliedContext;
```

Comment:
I would prefer to identity this as `user....` .

Reply:
k